scipy/stats/_stats.pyx

```diff
@@ -159,6 +159,7 @@ def _toint64(x):
         j = 1
         l = i + 1
 
+    i = -1
```

Comment:
This seems a subtle way to address the issue. At the very least it would need a comment.
Perhaps it would be clearer to change this line?
https://github.com/scipy/scipy/blob/6e6633102b8d65fd337f3ff598189c81fa4441ee/scipy/stats/_stats.pyx#L168 
```
result[perm[l-1]] = j
```

Reply:
Thank You for your advice! It's definitely clearer to change this line than the original one.

Commited enhanced version